alphagov/paas-cf
# [#117605669] Signing of Code for Production.

## What

We wish to PGP sign any merge commits in our paas-cf repository. Signing of commits is available through the command line, but not through the the web interface, so we must use the command line `git` client for all of these operations. The "Merge" button in the GitHub UI has a lot of functionality which we wish to preserve (such as ensuring that all status checks have passed successfully, generating commit meesages, allowing us to easily delete the branch after merging, etc) - We have replicated the necessary parts of this functionality within the code.

See corresponding [documentation PR](https://github.com/alphagov/paas-team-manual/pull/34) for more details.

## How to test
Pre-requisite: You will need to have your GPG goodies setup before you are able to review this PR - If you are already set up for using the credentials repo then you're good to go. 

For a regular PR made from a branch of a repository
* Checkout this branch of paas-cf
* Create a test repository of your own (it doesn't need to be in alphagov).
* Create a PR on your newly created repo
* Run `<path_to_paas-cf>/scripts/merge_pr.rb --pr <num>`
* You will be prompted by pinentry to enter your gpg password
* Watch in awe and wonder as everything works successfully

For a PR raised from a branch of a fork from our repo
* Create a fork from your repo
* Make some change in your forked repo
* Raise a PR from your fork against your original repo
* Run `<path_to_paas-cf>/scripts/merge_pr.rb --pr <num>`
* Watch in amazement as everything works successfully again

Finally...and **only if you're convinced that everything is working as expected**
* cd into the root of paas-cf
* use the Makefile target `make merge_pr PR=<num>` to merge in this PR itself.

## Who can merge this PR

Anyone but @alext or @jimconner
